UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneExporter.cs

```diff
@@ -93,39 +123,105 @@ public void SaveGLTFandBin(string path, string fileName)
 			gltfFile.Dispose();
 			binFile.Dispose();
 #else
-			gltfFile.Close();
-			binFile.Close();
+			gltfFile.Close ();
+			binFile.Close ();
 #endif
+			ExportImages (path);
+			
+		}
 
-			foreach (var image in _images)
-			{
-				Debug.Log(image.name);
-				var renderTexture = RenderTexture.GetTemporary(image.width, image.height);
-				Graphics.Blit(image, renderTexture);
-				RenderTexture.active = renderTexture;
-				var exportTexture = new Texture2D(image.width, image.height);
-				exportTexture.ReadPixels(new Rect(0, 0, renderTexture.width, renderTexture.height), 0, 0);
-				exportTexture.Apply();
-				File.WriteAllBytes(Path.Combine(path, image.name + ".png"), exportTexture.EncodeToPNG());
+		private void ExportImages (string outputPath)
+		{
+			for (int t = 0; t < _imageInfos.Count; ++t) {
+				var image = _imageInfos [t].texture;
+				int height = image.height;
+				int width = image.width;
+
+				switch (_imageInfos [t].textureMapType) {
+				case TextureMapType.MetallicGloss:
+					ExportMetallicGlossTexture (image, outputPath);
+					break;
+				case TextureMapType.Bump:
+					ExportNormalTexture (image, outputPath);
+					break;
+				default:
+					ExportTexture (image, outputPath);
+					break;
+				}
 			}
 		}
 
-		private SceneId ExportScene(string name, Transform[] rootObjTransforms)
+		private void ExportMetallicGlossTexture (Texture2D texture, string outputPath)
 		{
-			var scene = new Scene();
+			var destRenderTexture = new RenderTexture (texture.width, texture.height, 0, RenderTextureFormat.ARGB32, RenderTextureReadWrite.Linear);
+			var shader = Resources.Load ("MetalGlossChannelSwap", typeof(UnityEngine.Shader)) as UnityEngine.Shader;
+			var metalGlossMaterial = new UnityEngine.Material (shader);
 
-			if (ExportNames)
+			Graphics.Blit (texture, destRenderTexture, metalGlossMaterial);
+
+			var exportTexture = new Texture2D (texture.width, texture.height);	
+			exportTexture.ReadPixels (new Rect (0, 0, destRenderTexture.width, destRenderTexture.height), 0, 0);	
+			exportTexture.Apply ();
```

Comment:
I think this will do a CPU and GPU copy even though we just care about the CPU version here to write out. Is there a way to do this without creating an additional texture (like copying the bytes of the destRenderTexture into a file?)

Reply:
Looking around, it seems like this is the recommended pattern, add a comment explaining why we create another gpu copy of the data

Reply:
I'm not sure why this is the recommended pattern. Is there a better way to do this? I was following the same pattern as before.